options_filter.go

```diff
@@ -0,0 +1,37 @@
+package libp2p
+
+import (
+	"github.com/libp2p/go-libp2p-core/connmgr"
+	"github.com/libp2p/go-libp2p-core/control"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// FiltersConnectionGater is an adapter that turns multiaddr.Filter into a
+// connmgr.ConnectionGater. It's not intended to be used directly by the user,
+// but it can.
+type FiltersConnectionGater ma.Filters
```

Comment:
@raulk shouldn't this live somewhere else? Or is there a better implementation I should use in go-ipfs?

Reply:
(at the moment, this lives in the go-libp2p root package)

Reply:
Yeah, there's no reason to make this public: https://github.com/libp2p/go-libp2p/pull/946